src/Symfony/Component/HttpFoundation/ChainRequestMatcher.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpFoundation;
+
+/**
+ * ChainRequestMatcher compares a pre-defined set of checks against a Request instance.
```

Comment:
This should probably mention that it matches when all matchers are matching (this is a AND between them, not a OR)